### ISSUE TITLE
Deviations from the paper

### ISSUE THREAD

**Issue by hashbangCoder** [mjhucla/TF-mRNN#2]
Hi,

Thanks for the TF implementation.  I was going through the code and there seem to be a lot of differences between the paper and implementation? Eg: dropout,single-embedding layer,size of layers,etc. One in particular is the last layer, why not use a softmax directly [(this line)](https://github.com/mjhucla/TF-mRNN/blob/master/py_lib/tf_mrnn_model.py#L154) on the output of mrnn layer as is given in the paper? Why do you embed it back into the word space?

**Comment by mjhucla** [comment]
Thanks for the message. The shared-weight strategy of softmax is introduced in [this paper](https://arxiv.org/abs/1504.06692) and we find that for captioning it increases the performance and accelerates the training in practice. You can see more details and insight behind this strategy in the paper. 

I just re-implement the simplest version of the mRNN (there might be some small changes according to the paper) and you are free to explore other possible configurations of the model :)
